topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClient.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.topic.cruisecontrol;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.strimzi.api.kafka.model.topic.KafkaTopic;
+
+import java.net.http.HttpResponse;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Cruise Control REST API client.
+ * <br/><br/>
+ * The server runs one task execution at a time, additional 
+ * requests are queued up to {@code max.active.user.tasks}.
+ */
+public interface CruiseControlClient extends AutoCloseable {
+    /**
+     * HTTP request timeout in seconds.
+     */
+    long HTTP_REQUEST_TIMEOUT_SEC = 60;
+
+    /**
+     * Send a POST request to {@code topic_configuration} endpoint.
+     * This can be used to request replication factor changes.
+     * 
+     * @param kafkaTopics List of Kafka topics.
+     * @return The user task id.
+     */
+    String topicConfiguration(List<KafkaTopic> kafkaTopics);
```

Comment:
1. Do you really want to couple this interface to `KafkaTopic` like this? Or if this operation is just about RF change (and just happens to be implemented via the topic_configuration endpoint) why not take a `Map<String, Integer>` parameter?

Reply:
Well, the endpoint is defined as generic. Right now it only supports RF changes, but the goal is to support all topic configurations.

https://github.com/linkedin/cruise-control/wiki/REST-APIs#change-kafka-topic-configuration

IMO, KafkaTopic makes sense if we stay in the context of the Strimzi operators. Instead, we would need something different if we decide to have a general purpose CC client project. If you agree, I would leave it as it is right now, and we can discuss this point in a proposal about sharing the CC client.